protobuf/lib/src/protobuf/message_set.dart

```diff
@@ -74,12 +85,17 @@ abstract class MessageSet extends GeneratedMessage {
     mergeFromCodedBufferReader(CodedBufferReader(input), extensionRegistry);
   }
 
-  void _parseExtension(int typeId, List<int> message, ExtensionRegistry extensionRegistry) {
+  void _parseExtension(
+      int typeId, List<int> message, ExtensionRegistry extensionRegistry) {
     final ext = extensionRegistry.getExtension('MessageSet', typeId);
     if (ext == null) {
-      _fieldSet._ensureUnknownFields().addField(typeId, UnknownFieldSetField()..addLengthDelimited(message));
+      _fieldSet._ensureUnknownFields().addField(
+          typeId, UnknownFieldSetField()..addLengthDelimited(message));
     } else {
-      setExtension(ext, (ext.makeDefault!() as GeneratedMessage).deepCopy()..mergeFromBuffer(message));
+      setExtension(
+          ext,
+          (ext.makeDefault!() as GeneratedMessage).deepCopy()
```

Comment:
Doesn't `makeDefault` already create a fresh message? Are we sure we have to do deepcopy here?

Reply:
Ah - I think you want `subBuilder()` it should give you a fresh instance